src/nunit.analyzers/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzer.cs

```diff
@@ -43,22 +43,41 @@ private void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
             {
                 context.CancellationToken.ThrowIfCancellationRequested();
 
-                var arguments = attributeNode.ArgumentList.Arguments;
+                if (attributeNode.ArgumentList is AttributeArgumentListSyntax argumentList &&
+                    argumentList.Arguments.Count == 1 &&
+                    argumentList.Arguments.FirstOrDefault()?.Expression is LiteralExpressionSyntax literal &&
+                    literal.IsKind(SyntaxKind.StringLiteralExpression))
+                {
+                    if (HasMember(context, literal))
+                    {
+                        var stringConstant = literal.Token.ValueText;
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            CreateDescriptor(string.Format(TestCaseSourceUsageConstants.ConsiderNameOfInsteadOfStringConstantMessage, stringConstant)),
+                            literal.GetLocation()));
+                    }

```

Comment:
We should report another error here, I'd use a different id as it is a proper error.